### ISSUE TITLE
m_mesh non initialisé dans HyperionMainDriver

### ISSUE THREAD

**Issue by afreyermuth98** [bfovet/hyperion-lag#1]
Bonjour,

Au niveau de la classe HyperionMainDriver.cpp, notamment au niveau de la fonction run(), au départ de cette fonction on récupère des données via m_mesh qui semble être une UnstructuredGrid mais jamais initialisé, doit on la lier à ugrid dans la fonction load_mesh telle que m_mesh = ugrid ?

Cordialement.

**Comment by bfovet** [comment]
Oui. C'est un membre de la classe: `vtkSmartPointer<vtkUnstructuredGrid> m_mesh` qui doit être initialisé et utilisé dans `run()`.

**Comment by afreyermuth98** [comment]
Il faut donc bien écrire du code en dehors d'une zone de TODO pour cette initialisation ?

Du coup à la fin de la fonction load_mesh, j'ai fait m_mesh = ugrid, et j'ai m_mesh->GetNumberOfPoints() = 0 (contrairement a GetNumberOfCells() qui est bien non nul). Cela veut donc forcément dire que mon ugrid créé précédemment dans la boucle for à la fin de load_mesh n'est pas bon ? Dans ce cas, y-a-t-il une autre fonction que InsertNextCell à utiliser dans cette boucle ?

Cordialement.

**Comment by bfovet** [comment]
Non tu peux simplement faire l'initialisation au niveau du commentaire "Create a VTK unstructured grid": `m_mesh = vtkSmartPointer<vtkUnstructuredGrid>::New();`. Attention, il faut passer par des mécanismes de deep copy pour pouvoir faire un `m_mesh = ugrid` en VTK. Autant faire simple et éviter ce genre de choses... Je vois dans ton code que tu crées un `vtkPoints` mais tu n'en fais rien après.
